pkg/genericapiserver/storage_factory.go

```diff
@@ -57,6 +57,8 @@ type DefaultStorageFactory struct {
 
 	Overrides map[unversioned.GroupResource]groupResourceOverrides
 
+	DefaultResourcePrefixes map[unversioned.GroupResource]string
```

Comment:
Having customizable defaults and overrides separate (like in this PR) is actually cleanest, IMO. Otherwise the caller will have to figure out if the user's flags imply a default change. If they're split like that, we can leave the code interpreting the flags in genericapiserver.